packages/backend/src/services/ProjectService/ProjectService.ts

```diff
@@ -2133,6 +2149,33 @@ export class ProjectService {
                 await this.savedChartModel.getInfoForAvailableFilters(
                     savedQueryUuids,
                 );
+            const exploreCache: Record<string, Explore> = {};
+
+            const uniqueExplores = new Set<string>();
+            savedCharts.forEach((chart) => {
+                const key = `${chart.projectUuid}_${chart.tableName}`;
+                if (!uniqueExplores.has(key)) {
+                    uniqueExplores.add(key);
+                }
+            });
+
+            const explorePromises = Array.from(uniqueExplores).map(
+                async (key) => {
+                    const [projectUuid, tableName] = key.split('_');
+                    const explore = await this.getExplore(
```

Comment:
This await means you are running them sequentially, right? I think this will block before sending the next one. But since it returns promises, you should be able to run them in parallel, then await all of them on line 2174 with Promise.all. No?

I haven't tested this, so do let me know if I am missing something -- I could be wrong.

Reply:
You mean the `explorePromises` or this + the other `filterPromises`?

Reply:
I mean for explorePromises, you are awaiting each one, here, then you are awaiting all of them on line 2174, no? I don't think you need this await and I think it is causing them to run in series instead of in parallel.

Reply:
Ah I see your concern, makes sense! However, since the map method doesn't wait for the promises to resolve, all calls to getExplore are initiated almost simultaneously, those are then awaited in the `promise.all`

Reply:
Interesting. Not sure I follow what's going with needing the await, but that's fine with me. I do see what you are saying about map not waiting, so my concern is probably not valid.

Reply:
@joaoviana I see what you mean that the promise is actually the one from the anonymous function so all the anonymous functions fire in parallel.

but I think inlining this makes the behaviour a bit confusing. could you pull out the anonymous function like:

```typescript
const f = async (key) => {... await getExplore(..) ... }
explorePromises = Array.from(uniqueExplores).map(f)
```